File: app/src/main/java/org/mifos/selfserviceapp/models/Page.java
package org.mifos.selfserviceapp.models;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Rajan Maurya
 */
public class Page<T> {

    private int totalFilteredRecords;
    private List<T> pageItems = new ArrayList<T>();

    public List<T> getPageItems() {
        return pageItems;
    }

    public void setPageItems(List<T> pageItems) {
        this.pageItems = pageItems;
    }


    public int getTotalFilteredRecords() {
        return totalFilteredRecords;
    }

    public void setTotalFilteredRecords(int totalFilteredRecords) {
        this.totalFilteredRecords = totalFilteredRecords;
    }
}

File: core/model/src/commonMain/kotlin/org/mifos/mobile/core/model/entity/Page.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/mobile-mobile/blob/master/LICENSE.md
 */
package org.mifos.mobile.core.model.entity

/**
 * @author Rajan Maurya
 */
data class Page<T>(

    val totalFilteredRecords: Int = 0,
    val pageItems: List<T> = emptyList(),
)
